src/core/components/NewInstallButton/index.js

```diff
@@ -0,0 +1,336 @@
+/* @flow */
+/* global window */
+import makeClassName from 'classnames';
+import config from 'config';
+import * as React from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+
+import {
+  ADDON_TYPE_OPENSEARCH,
+  ADDON_TYPE_STATIC_THEME,
+  ADDON_TYPE_THEME,
+  DISABLED,
+  DOWNLOADING,
+  ENABLED,
+  ENABLING,
+  INSTALLED,
+  INSTALLING,
+  INSTALL_ACTION,
+  INSTALL_STARTED_ACTION,
+  UNINSTALLING,
+  UNKNOWN,
+} from 'core/constants';
+import translate from 'core/i18n/translate';
+import { findInstallURL } from 'core/installAddon';
+import log from 'core/logger';
+import { getThemeData } from 'core/themeInstall';
+import tracking, {
+  getAddonTypeForTracking,
+  getAddonEventCategory,
+} from 'core/tracking';
+import { isTheme } from 'core/utils';
+import AnimatedIcon from 'ui/components/AnimatedIcon';
+import Button from 'ui/components/Button';
+import Icon from 'ui/components/Icon';
+import type { AppState } from 'amo/store';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+import type { ReactRouterLocationType } from 'core/types/router';
+
+import './styles.scss';
+
+type Props = {|
+  addon: AddonType,
+  className?: string,
+  defaultInstallSource: string,
+  disabled: boolean,
+  location: ReactRouterLocationType,
+  puffy: boolean,
+  status: string,
+  // From `withInstallHelpers()`, see: `src/core/installAddon.js`.
```

Comment:
Is this comment correct? I see all props below this line getting set explicitly in `amo/Addon` and `disco/Addon`. I still see them set in `core/installAddon` as well so I'm a bit confused TBH.